extensions/amp-script/0.1/amp-script.js

```diff
@@ -109,19 +111,37 @@ export class AmpScript extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    return this.isExperimentOn_().then(on => {
-      if (!on) {
-        // Return rejected Promise to buildCallback() to disable component.
-        throw user().createError(TAG, `Experiment "${TAG}" is not enabled.`);
-      }
-    });
+    return this.isExperimentOn_()
+      .then(on => {
+        if (!on) {
+          // Return rejected Promise to buildCallback() to disable component.
+          throw user().createError('Experiment "%s" is not enabled.', TAG);
+        }
+        return getElementServiceForDoc(this.element, TAG, TAG);
+      })
+      .then(service => {
+        this.setService(service);
```

Comment:
Can this just be done in the previous then block?

Reply:
This will go away soon when we launch. :)